### ISSUE TITLE
The typing on the `compute` method of multi label metrics is wrong

### ISSUE THREAD

**Issue by Yann-CV** [Lightning-AI/torchmetrics#2704]
## 🐛 Bug

For multilabel metrics (at least: MultilabelROC), There is a mismatch between the returned value typing and the actual return type returned.

for instance:
MultilabelROC compute signature: `def compute(self) -> Tuple[Tensor, Tensor, Tensor]:`
while it is actually returning: Tuple[List[Tensor], List[Tensor], List[Tensor]].

Then type checking is complaining if one subclass MultilabelROC

### To Reproduce

```python
import torch
from torchmetrics.classification import (
    MultilabelROC,
)

m = MultilabelROC(num_labels=3)

preds = torch.randn(42, 3)
target = torch.randint(0, 2, (42, 3))

m.update(preds=preds, target=target)
result = m.compute()
print(f"the result is a {type(result)} of {len(result)}")
for r in result:
    print(f"type of an element: {type(r)}")
```

### Expected behavior

The typing is the correct one: `def compute(self) -> Tuple[List[Tensor], List[Tensor], List[Tensor]]:`

### Environment

- TorchMetrics version: 0.10.3
- Python & PyTorch Version (e.g., 1.0):  python 3.10.9 - torch 2.1.2
- Any other relevant information such as OS (e.g., Linux): Ubuntu Ubuntu 20.04.6 LTS

**Comment by Yann-CV** [comment]
I see that in 1.4.1 the typing is now the same as for the precision recall curve:
`Union[Tuple[Tensor, Tensor, Tensor], Tuple[List[Tensor], List[Tensor], List[Tensor]]]`

Still I do not really get why the Union would be required ?

**Comment by SkafteNicki** [comment]
Hi @Yann-CV,
If you look at the documentation it say that the output will either be a tuple of tensors or a tuple of list of tensors
https://lightning.ai/docs/torchmetrics/stable/classification/roc.html#multilabelroc
![image](https://github.com/user-attachments/assets/ef4b20af-8254-448c-967b-91c34ff45959)

The reason being that if you set the `thresholds` argument to something other than `None` (the default)
```python
m = MultilabelROC(num_labels=3)  # return tuple of lists
m = MultilabelROC(num_labels=3, thresholds=100)  # return tuple of tensors
```
when the thresholds is set to a specific number for all labels in multilabel when can compute the results more efficiently and compress the results into a single tensor of shape `[num_labels, num_thresholds]`.

Hope this explains the typing!

**Comment by Yann-CV** [comment]
@SkafteNicki Thanks for the feedback.
